File: src/main/java/game/Camera/ObstructionVisibility.java
package game.Camera;

import javafx.scene.Node;

public class ObstructionVisibility {

    public void setVisibility(Node node, Double gameScreenWidth, Double gameScreenHeight) {
        if (isVisible(node,gameScreenWidth,gameScreenHeight)) {
            node.setVisible(false);
        } else {
            node.setVisible(true);
        }
    }

    private boolean isVisible(Node node, Double gameScreenWidth, Double gameScreenHeight) {
        return node.getBoundsInParent().getMinX() < 0 ||
                gameScreenWidth < node.getBoundsInParent().getMaxX() ||
                node.getBoundsInParent().getMinY() < 0 ||
                gameScreenHeight < node.getBoundsInParent().getMaxY();
    }
}

File: src/main/java/game/Camera/ObstructionVisibility.kt
package game.Camera

import javafx.geometry.Rectangle2D
import java.util.*

class ObstructionVisibility {
    fun isVisible(node: Rectangle2D, gameScreenWidth: Double, gameScreenHeight: Double): Optional<Rectangle2D> {
        if (areNodeXsInBounds(node, gameScreenWidth) && areNodeYsInBounds(node, gameScreenHeight)) {
            return Optional.of(node)
        }
        return Optional.empty()
    }

    private fun areNodeXsInBounds(node: Rectangle2D, gameScreenWidth: Double): Boolean {
        return node.minX > 0 &&
                gameScreenWidth > node.maxX
    }

    private fun areNodeYsInBounds(node: Rectangle2D, gameScreenHeight: Double): Boolean {
        return node.minY > 0 &&
                gameScreenHeight > node.maxY
    }
}
